vault/expiration.go

```diff
@@ -1806,6 +1823,55 @@ func (m *ExpirationManager) deleteLockForLease(id string) {
 	m.lockPerLease.Delete(id)
 }
 
+func (m *ExpirationManager) sendEventByID(eventType logical.EventType, leaseID string) {
+	le, err := m.loadEntry(context.Background(), leaseID)
+	if err != nil {
+		m.logger.Warn("Error fetching lease information", "error", err)
+		return
+	}
+	m.sendEvent(eventType, le)
+}
+
+func (m *ExpirationManager) sendEvent(eventType logical.EventType, le *leaseEntry) {
+	data := &logical.EventData{}
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		m.logger.Warn("Error creating uuid", "error", err)
+		return
+	}
+	ns := namespace.RootNamespace
+	if le.namespace != nil {
+		ns = le.namespace
+	}
+	data.Id = id
+	data.EntityIds = []string{le.ClientToken}
+	structMap := map[string]interface{}{
```

Comment:
Should last_renewal_time only be populated if nonzero?  Should we include IssueTime (when it was created)?  What about RevokeErr?